File: Projects/05/Clojure/src/quadratic.clj
;;;; quadtratic.clj computes the quadtratic formula
;;;;
;;;; Input: three numbers
;;;; Precondition: a > 0
;;;; Output: the roots of the quadratic equation
;;;;
;;;; Completed by: AJ Vrieland (ajv234)
;;;; Date: 03/11/2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns quadratic)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; quadratic() computes the roots for a quadratic equation
;;; Receive: A, B, C numbers
;;; PRE: a > 0
;;; Return: a vector of the roots
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn quadratic [A B C]
  (def args (- (Math/pow B 2) (* 4 (* A C))))
  (def r1 (/ (- B args) (* 2 A)))
  (def r2 (/ (+ B args) (* 2 A)))
  (cond
    (< args 0) (vector 0 0)
    :else (vector r1 r2)
  )
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; -main
;;; Input: A, B, C from a quadratic equation
;;; Output: the roots of the formula
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn -main []
  (print "Enter A, B, C: ") (flush)
  (let
    [Qa (double (read))
     Qb (double (read))
     Qc (double (read))
     roots (quadratic Qa Qb Qc)
     root1 (get roots 0)
     root2 (get roots 1)]
     (printf (str "Root 1: \"%s\"; Root 2: \"%s\"\n") root1 root2)
     (println)

  )
)

File: Projects/05/Ruby/quadratic.rb
# quadratic.rb computes the quadratic roots of three numbers
#
# Completed by: AJ Vrieland (ajv234)
# Date: 03/11/2020
#
# Input: Three numbers, a, b, c
# Precondition: a must be positive
# Output: two quadratic roots of those numbers
#############################################################

# quadratic() takes in three numbers and returns the quadratic roots
# Receive: a, b, c
# PRE: a > 0
# Return: a vector of the quadratic roots
################################################################

def quadratic(a, b, c)
  args = (b**2 - 4.0*a*c)
  if args <= 0
    return [0, 0]
  elsif a == 0
    return [0, 0]
  else
    return [(b-Math.sqrt(args)) / 2 * a, (b+Math.sqrt(args)) / 2 * a]
  end
end

if __FILE__ == $0
  print "Enter A B C: ";
  qa = gets.to_i
  qb = gets.to_i
  qc = gets.to_i
  roots = quadratic(qa, qb, qc)

  if roots[0] == 0 && roots[1] == 0
    puts "Not Valid"
  else
    printf "\n The first root is: %f. The second root is: %f. \n", roots[0], roots[1]
  end
end
